### PR TITLE
Add recipe for program Genesis 4

### PR DESCRIPTION
**Pull request by klauer-xlight**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by klauer-xlight** [comment]
@conda-forge-admin, please ping conda-forge/help-c-cpp. This PR is ready for review. Thanks in advance!

File: recipes/genesis4/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+package:
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  url: https://github.com/svenreiche/Genesis-1.3-Version4/archive/v{{ version }}.tar.gz
+  sha256: c53ba86343bebd965ce3c358e3db13a4ebb3f47d6e0590427fb82a7593f004dc
+
+build:
+  number: {{ build }}
```
Comment (xhochy) [inline recipes/genesis4/meta.yaml:17]: Please leave the build number as a plain number.
```suggestion
  number: 0
```

File: recipes/genesis4/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,63 @@
+{% set name = "genesis4" %}
+{% set version = "4.6.4" %}
+
+{% set build = 0 %}
```
Comment (xhochy) [inline recipes/genesis4/meta.yaml:4]:
```suggestion
```

File: recipes/genesis4/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+
+build:
+  number: {{ build }}
+
+  # add build string so packages can depend on mpi variants explicitly:
+  # `pkg * mpi_mpich_*` for mpich
+  # `pkg * mpi_*` for any mpi
+
+  {% set mpi_prefix = "mpi_" + mpi %}
+  string: {{ mpi_prefix }}_h{{ PKG_HASH }}_{{ build }}
```
Comment (xhochy) [inline recipes/genesis4/meta.yaml:24]:
```suggestion
  string: {{ mpi_prefix }}_h{{ PKG_HASH }}_{{ PKG_BUILDNUM }}
```

File: recipes/genesis4/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+
+package:
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  url: https://github.com/svenreiche/Genesis-1.3-Version4/archive/v{{ version }}.tar.gz
+  sha256: c53ba86343bebd965ce3c358e3db13a4ebb3f47d6e0590427fb82a7593f004dc
+
+build:
```
Comment (xhochy) [inline recipes/genesis4/meta.yaml:16]:
```suggestion
build:
  skip: true  # [win]
```

**Review by xhochy — CHANGES_REQUESTED** [review]

File: recipes/genesis4/meta.yaml (hunk 1 above)
Comment (klauer-xlight) [reply recipes/genesis4/meta.yaml]: Resolved in [7dbf3f4](https://github.com/conda-forge/staged-recipes/pull/24644/commits/7dbf3f41f5545dcbc0f6e6cc567e7b6076fd6678)

File: recipes/genesis4/meta.yaml (hunk 3 above)
Comment (klauer-xlight) [reply recipes/genesis4/meta.yaml]: Resolved in [7dbf3f4](https://github.com/conda-forge/staged-recipes/pull/24644/commits/7dbf3f41f5545dcbc0f6e6cc567e7b6076fd6678)

**Comment by klauer-xlight** [comment]
Thanks for the review, @xhochy. I believe I've addressed your suggestions and the builds are still passing.

**Review by xhochy — APPROVED** [review]

**Pull request merged**